src/Nethermind/Nethermind.Serialization.Rlp/MyTxDecoder/OptimismTxDecoder.cs

```diff
@@ -0,0 +1,191 @@
+        }
+
+        stream.WriteByte((byte)transaction.Type);
+        stream.StartSequence(contentLength);
+
+        EncodePayload(transaction, stream);
+    }
+
+    public int GetLength(Transaction transaction, RlpBehaviors rlpBehaviors, bool forSigning = false, bool isEip155Enabled = false, ulong chainId = 0)
+    {
+        int txContentLength = GetPayloadLength(transaction);
+        int txPayloadLength = Rlp.LengthOfSequence(txContentLength);
+
+        bool isForTxRoot = rlpBehaviors.HasFlag(RlpBehaviors.SkipTypedWrapping);
+        int result = isForTxRoot
+                ? (1 + txPayloadLength)
+                : Rlp.LengthOfSequence(1 + txPayloadLength);
+        return result;
+    }
+
+    private static void DecodePayload(Transaction transaction, RlpStream rlpStream)
+    {
+        transaction.SourceHash = rlpStream.DecodeKeccak();
+        transaction.SenderAddress = rlpStream.DecodeAddress();
+        transaction.To = rlpStream.DecodeAddress();
+        transaction.Mint = rlpStream.DecodeUInt256();
+        transaction.Value = rlpStream.DecodeUInt256();
+        transaction.GasLimit = rlpStream.DecodeLong();
+        transaction.IsOPSystemTransaction = rlpStream.DecodeBool();
+        transaction.Data = rlpStream.DecodeByteArray();
+    }
+
+    private static void DecodePayload(Transaction transaction, ref Rlp.ValueDecoderContext decoderContext)
+    {
+        transaction.SourceHash = decoderContext.DecodeKeccak();
+        transaction.SenderAddress = decoderContext.DecodeAddress();
+        transaction.To = decoderContext.DecodeAddress();
+        transaction.Mint = decoderContext.DecodeUInt256();
+        transaction.Value = decoderContext.DecodeUInt256();
+        transaction.GasLimit = decoderContext.DecodeLong();
+        transaction.IsOPSystemTransaction = decoderContext.DecodeBool();
+        transaction.Data = decoderContext.DecodeByteArray();
+    }
+
+    private static void DecodeSignature(Transaction transaction, RlpStream rlpStream, RlpBehaviors rlpBehaviors)
+    {
+        ulong v = rlpStream.DecodeULong();
+        ReadOnlySpan<byte> rBytes = rlpStream.DecodeByteArraySpan();
+        ReadOnlySpan<byte> sBytes = rlpStream.DecodeByteArraySpan();
+        if (v == 0 && rBytes.IsEmpty && sBytes.IsEmpty) return;
+        transaction.Signature = SignatureBuilder.FromBytes(v + Signature.VOffset, rBytes, sBytes, rlpBehaviors) ?? transaction.Signature;
+    }
+
+    private static void DecodeSignature(Transaction transaction, ref Rlp.ValueDecoderContext decoderContext, RlpBehaviors rlpBehaviors)
+    {
+        ulong v = decoderContext.DecodeULong();
+        ReadOnlySpan<byte> rBytes = decoderContext.DecodeByteArraySpan();
+        ReadOnlySpan<byte> sBytes = decoderContext.DecodeByteArraySpan();
+        if (v == 0 && rBytes.IsEmpty && sBytes.IsEmpty) return;
+        transaction.Signature = SignatureBuilder.FromBytes(v + Signature.VOffset, rBytes, sBytes, rlpBehaviors) ?? transaction.Signature;
+    }
```

Comment:
From reading the code, it seems like Optimism transactions have no signature (we never encode it at least), yet `.DecodeX` methods move forward the Rlp stream/context forcing us to keep the code around.